tests/batched_curlp.rs

```diff
@@ -0,0 +1,170 @@
+
+    let mut expected_hashes = vec![];
+
+    let cases = include!("fixtures/curl_p/input_8019_output_243.rs");
+    assert_eq!(cases.len(), BATCH_SIZE);
+
+    for (input, output) in cases.iter() {
+        let input_trit_buf = TryteBuf::try_from_str(input).unwrap().as_trits().encode::<B>();
+
+        let expected_hash = TryteBuf::try_from_str(output).unwrap().as_trits().encode::<T1B1Buf>();
+
+        assert_eq!(expected_hash.len(), HASH_LENGTH);
+
+        batch_hasher.add(input_trit_buf);
+        expected_hashes.push(expected_hash);
+    }
+
+    for (index, hash) in batch_hasher.hash().enumerate() {
+        assert_eq!(expected_hashes[index], hash, "input {} failed", index);
+    }
+}
+
+#[test]
+fn batched_curl_p_broadcasted_t1b1_input_243_output_243() {
+    let tests = include!("fixtures/curl_p/input_243_output_243.rs");
+
+    for test in tests.iter() {
+        batched_curlp_broadcasted::<T1B1Buf, T1B1>(test.0, test.1);
+    }
+}
+
+#[test]
+fn batched_curl_p_broadcasted_t5b1_input_243_output_243() {
+    let tests = include!("fixtures/curl_p/input_243_output_243.rs");
+
+    for test in tests.iter() {
+        batched_curlp_broadcasted::<T5B1Buf, T5B1>(test.0, test.1);
+    }
+}
+
+#[test]
+fn batched_curl_p_broadcasted_t1b1_input_6561_output_243() {
+    let tests = include!("fixtures/curl_p/input_6561_output_243.rs");
+
+    for test in tests.iter() {
+        batched_curlp_broadcasted::<T1B1Buf, T1B1>(test.0, test.1);
+    }
+}
+
+#[test]
+fn batched_curl_p_broadcasted_t5b1_input_6561_output_243() {
+    let tests = include!("fixtures/curl_p/input_6561_output_243.rs");
+
+    for test in tests.iter() {
+        batched_curlp_broadcasted::<T5B1Buf, T5B1>(test.0, test.1);
+    }
+}
+
+#[test]
+fn batched_curl_p_broadcasted_t1b1_input_8019_output_243() {
+    let tests = include!("fixtures/curl_p/input_8019_output_243.rs");
+
+    for test in tests.iter() {
+        batched_curlp_broadcasted::<T1B1Buf, T1B1>(test.0, test.1);
+    }
+}
+
+#[test]
+fn batched_curl_p_broadcasted_t5b1_input_8019_output_243() {
+    let tests = include!("fixtures/curl_p/input_8019_output_243.rs");
+
+    for test in tests.iter() {
+        batched_curlp_broadcasted::<T5B1Buf, T5B1>(test.0, test.1);
+    }
+}
+
+#[test]
+fn batched_curl_p_broadcasted_t1b1_iota_go_json() {
+    let mut reader = BufReader::new(File::open("tests/fixtures/curl_p/curlp81-iota-go.json").unwrap());
+    let mut data = String::new();
+
+    reader.read_to_string(&mut data).unwrap();
+
+    let json: serde_json::Value = serde_json::from_str(&data).unwrap();
+    let cases = json.as_array().unwrap();
+
+    for case in cases.iter() {
+        let object = case.as_object().unwrap();
+        let input = object["in"].as_str().unwrap();
+        let output = object["hash"].as_str().unwrap();
+        batched_curlp_broadcasted::<T1B1Buf, T1B1>(input, output);
+    }
+}
+
+#[test]
+fn batched_curl_p_broadcasted_t5b1_iota_go_json() {
```

Comment:
Probably I am missing something, but aren't `batched_curl_p_broadcasted_t1b1_iota_go_json` and `batched_curl_p_broadcasted_t5b1_iota_go_json` exactly identical?

Reply:
The underlying `TritBuf` should use a different encoding.

Reply:
I see, now it seems fixed.